### PR TITLE
feat: customize my rule

### PR DESCRIPTION
**Pull request by toshuyu**
let's see how preview works

### CODE REVIEW

**Comment by toshuyu** [comment]
Your current specified code review guidelines are:
- function, method, variable name should be meaningful and not ambiguous

Following these guidelines, here is a preview of the code review in `java`:

Subject code:
```java
public class BadCodeExample {
    
    public static void main(String[] args) {
        ChildClass child = new ChildClass();
        child.doSomething();
        child.doSomethingElse();
    }
}

class ParentClass {
    public int someNumber = 10;
    public String someString = "Hello";

    public void doSomething() {
        System.out.println("Doing something in ParentClass");
        someNumber++;
        someString += " World";
        System.out.println(someString);
    }

    public void doSomethingElse() {
        for (int i = 0; i < someNumber; i++) {
            System.out.println("Count: " + i);
            if (i % 2 == 0) {
                System.out.println("Even number");
            } else {
                System.out.println("Odd number");
            }
        }
    }
}

class ChildClass extends ParentClass {
    public int someNumber = 5;

    @Override
    public void doSomething() {
        System.out.println("Doing something in ChildClass");
        someNumber *= 2;
        System.out.println("someNumber in ChildClass: " + someNumber);
    }
    
    @Override
    public void doSomethingElse() {
        for (int i = 0; i < someNumber; i++) {
            System.out.println("Count in child: " + i);
            if (i % 2 == 0) {
                System.out.println("Even number in child");
            } else {
                System.out.println("Odd number in child");
            }
        }
    }
}

```

Deputy's code review:
 I have reviewed the provided Pull Request and will provide feedback based on the tasks outlined by the engineer and software engineering best practices.

1. Function, method, and variable names:

The names used for functions, methods, and variables are mostly meaningful, but there is room for improvement. Here are some suggestions:

* Rename `ParentClass` to `BaseClass` or `SuperClass`, as it is not necessarily acting as a parent class in the inheritance hierarchy.
* Rename `ChildClass` to `SubClass`, which is a more common term for a class that inherits from another class.
* Rename `someNumber` and `someString` in both classes to more descriptive names, such as `parentSomeNumber`, `childSomeNumber`, `parentSomeString`, and `childSomeString`. This will make it clearer which variable belongs to which class.
* Rename `doSomething()` and `doSomethingElse()` methods in both classes to more descriptive names that reflect their functionality.

2. Encapsulation:

It is not a good practice to have public fields in a class. Instead, use private fields and provide public getter and setter methods if necessary. In this case, the `someNumber` and `someString` fields should be made private and accessed through getter and setter methods.

3. Inheritance:

In the provided code, both the parent and child classes have methods with the same name, but they do not override the parent methods. To ensure proper inheritance, use the `@Override` annotation to indicate that a method is intended to override a method in the parent class.

4. Code duplication:

There is some duplicated code in the `doSomethingElse()` methods in both classes. To avoid duplication, consider moving the common functionality to a separate method in the parent class and calling it from both child classes.

5. Code style:

Follow a consistent code style throughout the project. For example, use either all tabs or all spaces for indentation, and ensure that braces are placed consistently.

In summary, the provided code can be improved by:

* Renaming functions, methods, and variables to be more descriptive.
* Encapsulating fields using private access modifiers and providing getter and setter methods if necessary.
* Using the `@Override` annotation to ensure proper inheritance.
* Removing code duplication.
* Following a consistent code style.

By addressing these issues, the code will be more maintainable, easier to understand, and follow software engineering best practices.
